### PR TITLE
Less exceptions with runtime format string

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Save format strings for `Coordination::Exception`/`KeeperException`. Also, remove confusing constructors that were used incorrectly sometimes:
https://github.com/ClickHouse/ClickHouse/blob/39e481934c8cf99fe698c9b5d109d8b9ab2464c6/src/Common/ZooKeeper/IKeeper.h#L456-L457

and replace them with `fromPath` and `fromMessage` static methods.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 9b5d36f66b5b1ea27e2095ff7719b2f0b4a812d2 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53424/9b5d36f66b5b1ea27e2095ff7719b2f0b4a812d2/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by tavplubix** [comment]
Integration tests - `test_grpc_protocol/test.py::test_logs` will update the test just before merging

File: programs/keeper/Keeper.cpp [hunk 1]
```diff
@@ -114,15 +114,16 @@ void Keeper::createServer(const std::string & listen_host, const char * port_nam
 
         if (listen_try)
         {
-            LOG_WARNING(&logger(), "{}. If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to "
+            LOG_WARNING(&logger(), "Listen [{}]:{} failed: {}. If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, "
```
Comment (Avogar) [inline programs/keeper/Keeper.cpp:117]: I guess we need to delete `message` declared a few lines above

**Review by Avogar — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests are finally ok

Fuzzers are broken because of sending logs from CI to the cloud - https://github.com/ClickHouse/ClickHouse/pull/53528 
Functional tests do not start for the same reason

**Pull request merged**